src/uipath/_cli/_runtime/_runtime_factory.py

```diff
@@ -1,21 +1,53 @@
+import os
+from typing import List, Optional
+
 from uipath._cli._runtime._contracts import (
-    UiPathBaseRuntime,
     UiPathRuntimeContext,
     UiPathRuntimeFactory,
 )
-from uipath._cli._runtime._runtime import UiPathScriptRuntime
+from uipath._cli._runtime._runtime import UiPathScriptRuntime, get_user_scripts
+
 
+class UiPathScriptRuntimeFactory(
+    UiPathRuntimeFactory[UiPathScriptRuntime, UiPathRuntimeContext]
+):
+    """Factory for Python script runtimes."""
 
-def generate_runtime_factory() -> UiPathRuntimeFactory[
-    UiPathBaseRuntime, UiPathRuntimeContext
-]:
-    runtime_factory: UiPathRuntimeFactory[UiPathBaseRuntime, UiPathRuntimeContext] = (
-        UiPathRuntimeFactory(
+    def __init__(self):
+        super().__init__(
             UiPathScriptRuntime,
             UiPathRuntimeContext,
             context_generator=lambda **kwargs: UiPathRuntimeContext.with_defaults(
                 **kwargs
             ),
         )
-    )
-    return runtime_factory
+
+    def discover_all_runtimes(self) -> List[UiPathScriptRuntime]:
+        """Get a list of all available Python script runtimes."""
+        scripts = get_user_scripts(os.getcwd())
+
+        runtimes = []
+        for script_path in scripts:
+            runtime = self._create_runtime(script_path)
+            if runtime:
+                runtimes.append(runtime)
+
+        return runtimes
+
+    def get_runtime(self, entrypoint: str) -> Optional[UiPathScriptRuntime]:
+        """Get runtime for a specific Python script."""
+        if not os.path.isabs(entrypoint):
+            script_path = os.path.abspath(entrypoint)
+        else:
+            script_path = entrypoint
+
+        if not os.path.isfile(script_path) or not script_path.endswith(".py"):
+            return None
+
+        return self._create_runtime(script_path)
```

Comment:
As mentioned above, this should also have one of the three special functions defined.